### PR TITLE
Improve `RedundantInheritedCheck`

### PR DESCRIPTION
**Pull request by jgardn3r**
The redundant inherited check has a few issues:
* Quick fixes delete comments and compiler directives
* Methods in parent classes with different parameter names were ignored, triggering false positives

This PR addresses these issues.

### CODE REVIEW

File: CHANGELOG.md
Comment (Cirras) [inline CHANGELOG.md:34]:
```suggestion

### Fixed

- Overly permissive parsing rules around `string` and `file` types.
- False positives around `message` handler methods in `RedundantInheritedCheck`.
- Quick fixes removing comments and compiler directives in `RedundantInheritedCheck`.
```

**Review by Cirras — CHANGES_REQUESTED** [review]

File: delphi-checks/src/main/java/au/com/integradev/delphi/utils/RoutineUtils.java
Comment (Cirras) [inline delphi-checks/src/main/java/au/com/integradev/delphi/utils/RoutineUtils.java:105]: As far as I can tell, this change isn't relevant to improving `RedundantInheritedCheck`.

File: delphi-checks/src/test/java/au/com/integradev/delphi/checks/RedundantInheritedCheckTest.java
Comment (Cirras) [inline delphi-checks/src/test/java/au/com/integradev/delphi/checks/RedundantInheritedCheckTest.java:127]:
```suggestion
  void testOverridingMethodWithDifferentParameterNamesShouldNotAddIssues() {
```

**Review by Cirras — CHANGES_REQUESTED** [review]

File: delphi-checks/src/main/java/au/com/integradev/delphi/utils/RoutineUtils.java
Comment (jgardn3r) [reply delphi-checks/src/main/java/au/com/integradev/delphi/utils/RoutineUtils.java:105]: This change alters the way parameters are equated, i.e., parameter lists are now equal if the parameter names are different.

**Comment by Cirras** [comment]
Happy to approve after the merge conflict is resolved.

File: CHANGELOG.md
Comment (Cirras) [inline CHANGELOG.md:34]: Oops, missed one thing. The rule key doesn't include the word `Check`.

```suggestion
- Quick fixes removing comments and compiler directives in `RedundantInherited`.
- False positives around `message` handler methods in `RedundantInherited`.
```

**Review by Cirras — CHANGES_REQUESTED** [review]

**Comment by Cirras** [comment]
Also a suggestion for the commit messages:
- Improve `RedundantInherited` quick fixes around comments
- Fix FPs in `RedundantInherited`

File: CHANGELOG.md
Comment (Cirras) [inline CHANGELOG.md:34]:
```suggestion
- False positives around `message` handler methods in `RedundantInherited`.
- False positives when parameter names were mismatched to the inherited method in `RedundantInherited`.
```

**Review by Cirras — CHANGES_REQUESTED** [review]
This time for sure!

**Review by Cirras — APPROVED** [review]
🎉

**Pull request merged**
